tools/config/__init__.py

```diff
@@ -710,8 +711,23 @@ def _generate_bootloader_build(self, rom_start, rom_size):
             if part.minaddr() != rom_start:
                 raise ConfigException("bootloader executable does not "
                                       "start at 0x%x" % rom_start)
-            part_size = (part.maxaddr() - part.minaddr()) + 1
-            part_size = Config._align_ceiling(rom_start + part_size, self.sectors) - rom_start
+            if len(part.segments()) == 1:
+                part_size = (part.maxaddr() - part.minaddr()) + 1
+                part_size = Config._align_ceiling(rom_start + part_size, self.sectors) - rom_start
+            else:
+                # assume first segment is at start
+                # second at the end or outside ROM
+                # rest outside regular ROM
+                if part.segments()[0][0] != 0x0:
```

Comment:
This is not always true; ROM does not always start at `0x0`. This needs to be compared with `rom_start`.